kDrive/UI/Controller/Files/File List/InMemoryFileListViewModel.swift

```diff
@@ -25,7 +25,14 @@ class InMemoryFileListViewModel: FileListViewModel {
     private let realm: Realm
 
     override init(configuration: Configuration, driveFileManager: DriveFileManager, currentDirectory: File) {
-        if let realm = currentDirectory.realm {
+        // TODO: Refactor to explicit realm state
+        /// We expect the object to be live in this view controller, if not detached.
+        var currentDirectory = currentDirectory
+        if currentDirectory.isFrozen, let liveDirectory = currentDirectory.thaw() {
```

Comment:
I can do a full refactor, but it would change a lot of files since this is based on inheritance. This is a minimal patch for the next version of kDrive.

Reply:
Discussed, this is fine for now.